File: Examples/VisualBasic/AsposePDF/Bookmarks/GetBookmarks.vb
Imports System.IO
Imports System
Imports Microsoft.VisualBasic
Imports Aspose.Pdf

Namespace AsposePDF.Bookmarks
    Public Class GetBookmarks
        Public Shared Sub Run()
            ' ExStart:GetBookmarks
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_Bookmarks()

            ' Open document
            Dim pdfDocument As New Document(dataDir & Convert.ToString("GetBookmarks.pdf"))

            ' Loop through all the bookmarks
            For Each outlineItem As OutlineItemCollection In pdfDocument.Outlines
                Console.WriteLine(outlineItem.Title)
                Console.WriteLine(outlineItem.Italic)
                Console.WriteLine(outlineItem.Bold)
                Console.WriteLine(outlineItem.Color)
            Next
            ' ExEnd:GetBookmarks
        End Sub
    End Class
End Namespace

File: Examples/CSharp/AsposePDF/Bookmarks/GetBookmarks.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using System;

namespace CSharp.AsposePdf.Bookmarks
{
    public class GetBookmarks
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_Bookmarks();

            //open document
            Document pdfDocument = new Document(dataDir + "GetBookmarks.pdf");

            //loop through all the bookmarks
            foreach (OutlineItemCollection outlineItem in pdfDocument.Outlines)
            {
                Console.WriteLine(outlineItem.Title);
                Console.WriteLine(outlineItem.Italic);
                Console.WriteLine(outlineItem.Bold);
                Console.WriteLine(outlineItem.Color);
            }

        }
    }
}